onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -368,121 +628,185 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
-                params.StockhamParams.ResourceLoopList.back().Strides = temporaryStrides;
+        barriers[1] = CD3DX12_RESOURCE_BARRIER::Transition(
+            outputResource,
+            D3D12_RESOURCE_STATE_COMMON,
+            D3D12_RESOURCE_STATE_UNORDERED_ACCESS
+        );
 
-                auto& resource = params.StockhamParams.ResourceLoopList.back().Resource;
-                ORT_THROW_IF_FAILED(context->AllocateTemporaryData(temporaryBufferByteSize, &resource));
-            }
+        commandList->ResourceBarrier(2, barriers);
 
-            // Add output resource
-            params.StockhamParams.ResourceLoopList.push_back({});
-            params.StockhamParams.ResourceLoopList.back().Resource = outputResource;
-            params.StockhamParams.ResourceLoopList.back().Sizes = reshapedOutputSize;
-            params.StockhamParams.ResourceLoopList.back().Strides = reshapedOutputStrides;
-            params.StockhamParams.OutputIndex = static_cast<uint32_t>(params.StockhamParams.ResourceLoopList.size() - 1);
+        // Set the root signature and pipeline state
+        commandList->SetComputeRootSignature(m_bluesteinChirpRootSignature.Get());
+        commandList->SetPipelineState(m_bluesteinChirpPipelineState.Get());
 
-            // Add the temporary after output incase of odd number of passes
-            if (oscillateFirstOutputThenTemporary)
-            {
-                params.StockhamParams.ResourceLoopList.push_back({});
-                params.StockhamParams.ResourceLoopList.back().Sizes = temporarySize;
-                params.StockhamParams.ResourceLoopList.back().Strides = temporaryStrides;
+        // Create ZChirp and B Tensors
+        BluesteinZChirpShaderConstants constants = {};
+        constants.DFTLength = dftParams.DFTLength;
+        constants.IsInverse = isInverse;
+
+        auto totalElementCount =
+            std::accumulate(bluesteinZChirpParams.B.Sizes.begin(),
+                            bluesteinZChirpParams.B.Sizes.end(),
+                            1,
+                            std::multiplies<uint32_t>());
+        constants.ElementCount = totalElementCount / bluesteinZChirpParams.B.Sizes[3];
+
+        std::array<ID3D12Resource*, 2> uav_resources = { zchirp_resource, b_resource };
+        Dispatch(uav_resources, constants, commandList);
+
+        DFTParameters fft_params = {};
+        fft_params.Type = DFTType::Stockham;
+        fft_params.BluesteinZChirpParams = {};
+        fft_params.DFTLength = bluesteinZChirpParams.AFFT.Sizes[1];
+
+        // Create BFFT Tensors
+        fft_params.StockhamParams = bluesteinZChirpParams.BFFTParams;
+        StockhamFFT(fft_params, false, 0 /*chirpLength*/, 1 /*scale*/, commandList);
+
+        // Create AFFT Tensors
+        fft_params.StockhamParams = bluesteinZChirpParams.AFFTParams;
+        StockhamFFT(fft_params, false, 0 /*chirpLength*/, 1 /*scale*/, commandList);
+
+        // Should include the bfft tensor as the window function
```

Comment:
```suggestion
        // Should include the BFFT tensor as the window function
```

I really wish GitHub and per-word suggestions rather than just whole line suggestions.